stripe.go

```diff
@@ -485,6 +370,107 @@ func (s *BackendConfiguration) ResponseToError(res *http.Response, resBody []byt
 	return stripeErr
 }
 
+// SetMaxNetworkRetries sets max number of retries on failed requests
+func (s *BackendConfiguration) SetMaxNetworkRetries(maxNetworkRetries int) {
+	s.MaxNetworkRetries = maxNetworkRetries
+}
+
+// Checks if an error is a problem that we should retry on. This includes both
+// socket errors that may represent an intermittent problem and some special
+// HTTP statuses.
+func (s *BackendConfiguration) shouldRetry(err error, resp *http.Response, numRetries int) bool {
+	if numRetries >= s.MaxNetworkRetries {
+		return false
+	}
+
+	if err != nil {
+		return true
+	}
+
+	if resp.StatusCode == http.StatusConflict {
+		return true
+	}
+	return false
+}
+
+// Backends are the currently supported endpoints.
+type Backends struct {
+	API, Uploads Backend
+	mu           sync.RWMutex
+}
+
+// Printfer is an interface to be implemented by Logger.
+type Printfer interface {
+	Printf(format string, v ...interface{})
+}
+
+// SupportedBackend is an enumeration of supported Stripe endpoints.
+// Currently supported values are "api" and "uploads".
+type SupportedBackend string
+
+//
+// Public functions
+//
+
+// Bool returns a pointer to the bool value passed in.
+func Bool(v bool) *bool {
+	return &v
+}
+
+// BoolValue returns the value of the bool pointer passed in or
+// false if the pointer is nil.
+func BoolValue(v *bool) bool {
+	if v != nil {
+		return *v
+	}
+	return false
+}
+
+// Float64 returns a pointer to the float64 value passed in.
+func Float64(v float64) *float64 {
+	return &v
+}
+
+// Float64Value returns the value of the float64 pointer passed in or
+// 0 if the pointer is nil.
+func Float64Value(v *float64) float64 {
+	if v != nil {
+		return *v
+	}
+	return 0
+}
+
+// GetBackend returns the currently used backend in the binding.
+func GetBackend(backend SupportedBackend) Backend {
+	switch backend {
+	case APIBackend:
+		backends.mu.RLock()
+		ret := backends.API
+		backends.mu.RUnlock()
+		if ret != nil {
+			return ret
+		}
+		backends.mu.Lock()
+		defer backends.mu.Unlock()
+		backends.API = &BackendConfiguration{Type: backend, URL: apiURL, HTTPClient: httpClient}
+		return backends.API
+
+	case UploadsBackend:
+		backends.mu.RLock()
+		ret := backends.Uploads
+		backends.mu.RUnlock()
+		if ret != nil {
+			return ret
+		}
+		backends.mu.Lock()
+		defer backends.mu.Unlock()
+		backends.Uploads = &BackendConfiguration{Type: backend, URL: uploadsURL, HTTPClient: httpClient}
+		return backends.Uploads
+	}
+
+	return nil
+}
+
 // FormatURLPath takes a format string (of the kind used in the fmt package)
```

Comment:
minor but it should be before `GetBackend`

Reply:
Great catch! Fixed.